pkg/utils/kubernetes/health/checker/checker.go

```
@@ -506,3 +536,106 @@ func (h *HealthChecker) checkControllerInstallationConditions(
+	managedResources []resourcesv1alpha1.ManagedResource,
+	filterFunc func(resourcesv1alpha1.ManagedResource) bool,
+) *gardencorev1beta1.Condition {
+	var (
+		prometheuses []*monitoringv1.Prometheus
+		tasks        []flow.TaskFn
+	)
+
+	for _, managedResource := range managedResources {
+		if !filterFunc(managedResource) || managedResource.Annotations[resourcesv1alpha1.Ignore] == "true" {
+			continue
+		}
+
+		for _, resource := range managedResource.Status.Resources {
+			if resource.Kind == monitoringv1.PrometheusesKind {
+				prometheus := &monitoringv1.Prometheus{ObjectMeta: metav1.ObjectMeta{Namespace: resource.Namespace, Name: resource.Name}}
+				prometheuses = append(prometheuses, prometheus)
+			}
+		}
+	}
+
+	// Guarantee failed conditions are returned in a stable order to avoid too many writes to condition statuses.
+	slices.SortFunc(prometheuses, func(i, j *monitoringv1.Prometheus) int {
+		if i.Namespace == j.Namespace {
+			return strings.Compare(i.Name, j.Name)
+		}
+		return strings.Compare(i.Namespace, j.Namespace)
+	})
+
+	conditions := make([]*gardencorev1beta1.Condition, len(prometheuses))
+	for i, prometheus := range prometheuses {
+		tasks = append(tasks, func(ctx context.Context) error {
+			conditions[i] = func() *gardencorev1beta1.Condition {
+				if err := h.reader.Get(ctx, client.ObjectKeyFromObject(prometheus), prometheus); err != nil {
+					if apierrors.IsNotFound(err) {
+						return ptr.To(v1beta1helper.FailedCondition(h.clock, h.lastOperation, h.conditionThresholds, condition, "PrometheusHealthCheckError", fmt.Sprintf("Prometheus \"%s/%s\" not found", prometheus.Namespace, prometheus.Name)))
+					}
+
+					return ptr.To(v1beta1helper.NewConditionOrError(h.clock, condition, nil, fmt.Errorf("failed checking Prometheus \"%s/%s\": %w", prometheus.Namespace, prometheus.Name, err)))
+				}
+
+				return h.CheckPrometheus(ctx, condition, prometheus)
+			}()
+			return nil
+		})
+	}
+
+	_ = flow.Parallel(tasks...)(ctx)
```

Why do we ignore the error here?

---

Now that I see this comment, I think this is wrong. Thanks!

Originally, we did it like this because `tasks` is a slice of functions that always return `nil`, so `flow.Parallel()` should return no error either, as far as I understand. But I guess this could change any time and returning the error won't hurt. Thank you, I will address this too.